### PR TITLE
Migrate to rollup

### PR DESCRIPTION
**Pull request by ilajosmanov**

### CODE REVIEW

**Comment by dangreen** [comment]
@ilajosmanov Please rebase it.

File: package.json [hunk 1]
```diff
@@ -14,15 +14,12 @@
     "charts component"
   ],
   "license": "MIT",
-  "main": "dist/vue-google-charts.common.js",
+  "main": "dist/vue-google-charts.cjs.js",
```
Comment (dangreen) [inline package.json:17]:
```suggestion
  "main": "dist/vue-google-charts.cjs",
```

Need to use `.cjs` extension. It's important for [conditional exports](https://nodejs.org/api/packages.html#conditional-exports)

File: package.json [hunk 2]
```diff
@@ -14,15 +14,12 @@
...
   "unpkg": "dist/vue-google-charts.browser.js",
-  "module": "index.js",
+  "module": "dist/vue-google-charts.esm.js",
```
Comment (dangreen) [inline package.json:19]:
```suggestion
  "module": "dist/vue-google-charts.mjs",
```

for consistency

File: rollup.config.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import commonjs from '@rollup/plugin-commonjs';
+import { nodeResolve } from '@rollup/plugin-node-resolve';
+
+import pkg from './package.json';
+
+const commonPlugins = [
+  replace({
+    preventAssignment: true,
+    values: {
+      VERSION: JSON.stringify(pkg.version)
```
Comment (dangreen) [inline rollup.config.js:12]: Is it really needed?

File: rollup.config.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    },
+    plugins: [
+      ...commonPlugins,
+      vue({ template: { optimizeSSR: true } })
+    ]
+  },
+  {
+    input: 'src/wrapper.js',
+    output: {
+      format: 'iife',
```
Comment (dangreen) [inline rollup.config.js:43]:
```suggestion
      format: 'umd',
```

?

File: rollup.config.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+      format: 'cjs',
+      file: pkg.main
+    },
+    plugins: [
+      ...commonPlugins,
+      vue({ template: { optimizeSSR: true } })
+    ]
+  },
+  {
+    input: 'src/wrapper.js',
```
Comment (dangreen) [inline rollup.config.js:41]:
```suggestion
    input: 'src/umd.js',
```

?

File: src/components/GChart.vue [hunk 6]
```diff
@@ -108,6 +108,7 @@ export default {
         return new google.visualization[type](el)
       }
       const fn = this.createChart || createChart
+      console.log(fn);
```
Comment (dangreen) [inline src/components/GChart.vue:111]: ?

**Review by dangreen — CHANGES_REQUESTED** [review]

File: rollup.config.js (hunk 3 above)
Comment (ilajosmanov) [reply rollup.config.js:12]: The version from package.json is already used in Vue.plugin. I wanna save this info for the package soon

Comment (dangreen) [reply rollup.config.js:12]: Then let's make it b more convenient way?
```
'import.meta.env.VERSION': JSON.stringify(pkg.version) 
```

(it's like modern alternative of `process.env`, https://dev.to/danawoodman/storing-environment-variables-in-sveltekit-2of3)

File: rollup.config.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+const commonPlugins = [
+  replace({
+    preventAssignment: true,
+    values: {
+      'import.meta.env.VERSION': JSON.stringify(pkg.version)
+    }
+  }),
+]
+
+export default [
```
Comment (dangreen) [inline rollup.config.js:17]: You forgot `external` option, to not include dependencies in bundle. It requiers `nodeResolve` plugin.

**Review by dangreen — CHANGES_REQUESTED** [review]
What about swc?

File: rollup.config.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
+  import vue from 'rollup-plugin-vue';
+  import swc from 'rollup-plugin-swc';
+  import replace from "@rollup/plugin-replace";
+  import commonjs from '@rollup/plugin-commonjs';
+  import { nodeResolve } from '@rollup/plugin-node-resolve';
```
Comment (dangreen) [inline rollup.config.js:5]: unnecessary indent

File: rollup.config.js [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+  },
+  {
+    external,
+    input: 'src/index.js',
+    output: {
+      format: 'cjs',
+      file: pkg.main
+    },
+    plugins: [
+      vue({ template: { optimizeSSR: true } }),
```
Comment (dangreen) [inline rollup.config.js:54]: lets move it to `plugins()`. We can add second optional parameter to it to pass this options.

**Review by dangreen — CHANGES_REQUESTED** [review]

**Review by dangreen — APPROVED** [review]

**Pull request merged**